lib/sles4sap_publiccloud.pm

```diff
@@ -531,20 +533,40 @@ sub check_takeover {
 
 sub enable_replication {
     my ($self, %args) = @_;
-    croak("Argument <site_name> missing") unless $args{site_name};
+    croak('enable_replication [ERROR] Argument <site_name> missing') unless $args{site_name};
     my $hostname = $self->{my_instance}->{instance_id};
-    die("Database on the fenced node '$hostname' is not offline") if ($self->is_hana_database_online);
-    die("System replication '$hostname' is not offline") if ($self->is_primary_node_online);
+    my $remote_host;
+    my $sr_mode;
+    my $op_mode;
+    my $instance_id;
+    die("enable_replication [ERROR] Database on the fenced node '$hostname' is not offline") if ($self->is_hana_database_online);
+    die("enable_replication [ERROR] System replication '$hostname' is not offline") if ($self->is_primary_node_online);
 
     my $topology = $self->get_hana_topology();
-    foreach (qw(vhost remoteHost srmode op_mode)) { die "Missing '$_' field in topology output" unless defined(%$topology{$hostname}->{$_}); }
+    for my $site (keys %{$topology->{'Site'}}) {
```

Comment:
Do we need to iterate over all Sites in the parsed topology even if we only care `$args{site_name}`. Can we just check for `$args{site_name}` key presence in the topology?